From fc592ca27fd5999cdd57ae22e0f9abd7c91cd7c4 Mon Sep 17 00:00:00 2001
From: Corey Minyard <cminyard@mvista.com>
Date: Wed, 23 Apr 2014 14:36:47 -0500
Subject: [PATCH 2/3] Fix hang processing empty free list

If zone data was not used, it would be all zeros, including the free
list.  This could result in issues accessing the free list.  So ignore
NULL free list head.
---
 kdump-tool.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/kdump-tool.c b/kdump-tool.c
index 69d2b03..35175c7 100644
--- a/kdump-tool.c
+++ b/kdump-tool.c
@@ -746,6 +746,9 @@ process_free_list(struct kdt_data *d, unsigned int order, unsigned char *list)
 	if (rv == -1)
 		return -1;
 
+	if (head_next == 0)
+		return 0;
+
 	next = head_next;
 	for (;;) {
 		GElf_Addr page_offset = next - d->page_lru_offset;
-- 
1.8.3.1

